sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -274,14 +221,23 @@ def convert_feature(fx):
                              'len(feature_names) = {0}, got {1}.'
                              .format(len(feature_names), i))
 
-    # compute averaged predictions
-    pd_results = Parallel(n_jobs=n_jobs, verbose=verbose)(
-        delayed(partial_dependence)(estimator, X, fxs,
-                                    response_method=response_method,
-                                    method=method,
-                                    grid_resolution=grid_resolution,
-                                    percentiles=percentiles)
-        for fxs in features)
+    # compute predictions
+    if plot_type == 'pdp':
```

Comment:
We need the name of the common parameters to be the same and then we can simplify like:

```python
params = {
    "est": estimator, "X": X, ...
}
if plot_type == partial_dependence:
    params["method"] = method
    pd_results = Parallel(n_jobs=n_jobs, verbose=verbose)(
            delayed(plot_type)(**params) for fxs in features
    )
```